### DISCUSSION TITLE
Tiled Plugin Doesn't Draw the Map

### DISCUSSION THREAD

**Discussion by cemalgnlts** [excaliburjs/Excalibur#2716] [Q&A] [answered]
Hi,

I discovered Excalibur today and wanted to add a TileMap nothing appears on the screen.

**main.js**:
```js
import * as ex from "/libs/excalibur.min.js";
import tiled from "https://cdn.skypack.dev/@excaliburjs/plugin-tiled";

// New game
const game = new ex.Engine({ width: 500, height: 400 });

// Create a new TiledMapResource loadable
const tiledMap = new tiled.TiledMapResource("/assets/map/level1.tmx");

// Create a loader and reference the map
const loader = new ex.Loader([tiledMap]);

// Start the game (starts the loader)
game.start(loader).then(function() {
  console.log("Game loaded");
  tiledMap.addTiledMapToScene(game.currentScene);
});
```

**/assets/map/level1.tmx**:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<map version="1.10" tiledversion="1.10.1" orientation="orthogonal" renderorder="right-down" width="10" height="5" tilewidth="64" tileheight="64" infinite="0" nextlayerid="2" nextobjectid="1">
 <tileset firstgid="1" name="Tile-Sets_64x64" tilewidth="64" tileheight="64" tilecount="30" columns="6">
  <image source="../Tile-Sets_64x64.png" width="384" height="320"/>
 </tileset>
 <layer id="1" name="foreground" width="10" height="5">
  <data encoding="csv">
19,14,14,14,14,14,14,14,14,7,
9,0,0,0,0,0,0,0,0,7,
9,0,0,0,0,0,0,0,0,7,
9,0,0,0,0,0,0,0,0,7,
25,2,2,2,2,2,2,2,2,26
</data>
 </layer>
</map>
```
![Screenshot 2023-08-01 16 14 02](https://github.com/excaliburjs/Excalibur/assets/45357531/22e3dd93-c439-4c7b-8b8b-779d303370f3)

**Comment by cemalgnlts** [comment, marked as answer]
I solved the problem but I can't describe it fully.

I had problems using excalibur and tiled plugin without using any packer, there was no error, but there was no map drawing, instead I started using unpkg versions for the browser (no modules required).

**Reply by eonarheim** [reply]
@cemalgnlts Thanks for the discussion!

I've seen this type of thing before when bundlers mangle class names. Excalibur and the Tiled plugin use `instanceof` checks internally at runtime, and some bundlers will obfuscate/minify the class names confusing the library.

We have an issue that links to a few discussions about it https://github.com/excaliburjs/Excalibur/issues/2436

We are still looking at solutions to be resistant to class name mangling

**Reply by cemalgnlts** [reply]
Thanks for the info, the problem makes more sense now.
